apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.UnitTests/ConnectionStringBuilderTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Google.Cloud.Spanner.V1;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class ConnectionStringBuilderTests
+    {
+        [Fact]
+        public void DataSourceParsing()
+        {
+            var connectionStringBuilder =
+                new SpannerConnectionStringBuilder(
+                    "Data Source=projects/project1/instances/instance1/databases/database1");
+            Assert.Equal("project1", connectionStringBuilder.Project);
+            Assert.Equal("instance1", connectionStringBuilder.SpannerInstance);
+            Assert.Equal("database1", connectionStringBuilder.SpannerDatabase);
+        }
+
+        [Fact]
+        public void InstanceOnlyDataSource()
+        {
+            var connectionStringBuilder =
+                new SpannerConnectionStringBuilder("Data Source=projects/project1/instances/instance1");
+            Assert.Equal("project1", connectionStringBuilder.Project);
+            Assert.Equal("instance1", connectionStringBuilder.SpannerInstance);
+            Assert.Null(connectionStringBuilder.SpannerDatabase);
+        }
+
+        [Fact]
+        public void BadDataSourceThrows1()
```

Comment:
I'd actually convert all of these into a single test using `[InlineData("bad connection string here")]` and then just put a comment next to each one explaining in what way it's bad. (It's a shame `InlineData` doesn't have a `Description` property or similar, but I think a comment is at least good enough.)